src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -632,6 +632,94 @@ private void CleanUp()
         #endregion IDisposable Members
     }
 
+    /// <summary>
+    /// Implements ConvertTo-CliXml command.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2243:AttributeStringLiteralsShouldParseCorrectly")]
+    [Cmdlet(VerbsData.ConvertTo, "CliXml", HelpUri = "")]
+    public sealed class ConvertToClixmlCommand : PSCmdlet
+    {
+        #region Parameters
+
+        /// <summary>
+        /// Gets or sets input objects to be converted to CliXml object.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public PSObject InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets depth of serialization.
+        /// </summary>
+        [Parameter]
+        [ValidateRange(1, int.MaxValue)]
+        public int Depth { get; set; } = 1;
+
+        #endregion Parameters
+
+        #region Private Members
+
+        private readonly List<PSObject> _inputObjectBuffer = new();
+
+        #endregion Private Members
+
+        #region Overrides
+
+        /// <summary>
+        /// Process record.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            _inputObjectBuffer.Add(InputObject);
+        }
+
+        /// <summary>
+        /// End Processing.
+        /// </summary>
+        protected override void EndProcessing()
+        {
+            if (_inputObjectBuffer.Count > 0)
+            {
+                // If single input object don't wrap in array
+                object source = _inputObjectBuffer.Count == 1
+                    ? _inputObjectBuffer[0]
+                    : _inputObjectBuffer.ToArray();
+                
+                WriteObject(PSSerializer.Serialize(source, Depth));
+            }
+        }
+
+        #endregion Overrides
+    }
+
+    /// <summary>
+    /// Implements ConvertFrom-CliXml command.
+    /// </summary>
+    [Cmdlet(VerbsData.ConvertFrom, "CliXml", SupportsPaging = true, HelpUri = "")]
+    public sealed class ConvertFromClixmlCommand : PSCmdlet
+    {
+        #region Parameters
+
+        /// <summary>
+        /// Gets or sets input object which is written in CliXml format.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true)]
+        public string InputObject { get; set; }
```

Comment:
Maybe PSObject?

Reply:
Wasn't sure about this one. To deserialize you need a string so didn't see why we'd keep it as PSObject.

Reply:
@mklement0 What do you think?

Reply:
@ArmaanMcleod, thanks for tackling this and the `Show-Markdown` issue. 
@iSazonov, using `string` makes sense to me, and it's also what `ConvertFrom-Json` and `ConvertFrom-StringData` do. Do you have specific concerns?

Reply:
@mklement0 I remember https://github.com/PowerShell/PowerShell/issues/4242 So it would be great to have you conclusion about the parameter and how the cmdlet process it.

Reply:
@iSazonov, by using `string` we avoid conceptual ambiguity and clearly communicate that the cmdlet can only accept _multiple_ input objects (strings) via the _pipeline_, as most of the existing cmdlets do. While for some cmdlets the ability to pass multiple objects via an array directly to `-InputObject` is important for performance reasons, I don't think we need to worry about it here: I expect the typical use case to be a _single_ input string. (Multiple input strings often come from lazy use of `Get-Content` rather than `Get-Content -Raw` as input, but the file-input case is covered by `Import-CliXml`).

Reply:
Wouldn't there be a problem with aggressive conversion to a string?

> Multiple input strings often come from lazy use of Get-Content

Original scenario was `Get-WhatsNew | Show-Markdown` where first cmdlet outputs an array of strings.

Reply:
@iSazonov, you're thinking of a specific `Show-Markdown` use case; I can't think of an analogous scenario for `ConvertFrom-CliXml`, whose input I'd typically expect to be a single string emitted by `ConvertTo-CliXml` (as noted, with _file_ input you'd just use `Import-CliXml`).

Reply:
Ah, sorry :-) There is another PR for Show-Markdown. My comment is in the wrong place.

Reply:
@iSazonov  👍, easy to get confused.
On an unrelated note: at least at first glance it looks like `Import-CliXml`'s `-IncludeCount` switch does nothing useful and always outputs a dummy result (a `[psobject]` that wraps `[double]` `0.0` and stringifies to `Unknown total count`). Do you know the backstory? If my analysis is correct, at least the docs should be updated.

Reply:
@mklement0 I don't know the history. I see Engine has formal support for "paging" parameters (like common and dynamic parameters). Only Import-CliXml has one line code to check for "paging" parameter - IncludeTotalCount - with zero values. This cmdlet definitely doesn't assign SupportsPaging to true in the CmdletAttribute and it's dead code. We could remove it.

Reply:
I was also thinking if it's even worth including `SupportsPaging` for these new CliXml cmdlets. I included it for consistency but honestly `IncludeCount` is totally useless right now, and `-First` and `-Skip` can be achieved with `Select-Object`. I'm not even convinced the performance is better since `SupportsPaging` still requires so much complicated code to filter. I could be totally wrong here since I havn't used `SupportsPaging` before, probably need say from others who have more background.

Reply:
Obviously, we have no need for SupportsPaging. Please remove.
(We could discuss the feature and its history in new issue if you have an interest.)

Reply:
Done